enhancements/network/configure-ovs-alternative.md

```diff
@@ -0,0 +1,498 @@
+  - @jcaamano
+  - @trozet
+approvers: # A single approver is preferred, the role of the approver is to raise important questions, help ensure the enhancement receives reviews from all applicable areas/SMEs, and determine when consensus is achieved such that the EP can move forward to implementation.  Having multiple approvers makes it difficult to determine who is responsible for the actual approval.
+  - TBD
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - TBD
+creation-date: 2023-06-29
+last-updated: 2023-07-13
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/OPNET-265
+see-also:
+  - https://docs.google.com/document/d/1Zp1J2HbVvu-v4mHc9M594JLqt6UwdKoix8xkeSzVL98/edit?usp=sharing
+  - enhancements/network/baremetal-ipi-network-configuration.md
+replaces:
+  - NA
+superseded-by:
+  - NA
+---
+
+# Configure-ovs Alternative
+
+## Summary
+
+There are some significant limitations to the existing configure-ovs.sh design
+in OVNKubernetes. This enhancement is an alternative method of configuring the
+br-ex bridge needed by OVNK using NMState that will address most, if not all,
+of the problems with the existing implementation.
+
+## Motivation
+
+There are a few problems with configure-ovs.sh:
+
+* It is implemented in Bash, which makes it fragile and difficult to test.
+* It is guessing at what the deployer wants for their bridge configuration,
+  and it is awkward to override when it guesses wrong.
+* It has customer-specific logic because there is no alternative for us to
+  support advanced use cases.
+* It is incompatible with day 2 changes to the bridge using things like the
+  Kubernetes-NMState operator because the operator and script configurations
+  will overwrite each other.
+
+### User Stories
+
+* As an OpenShift administrator, I want the OVNKubernetes bridge on a different
+  interface from the node IP.
+
+* As an OpenShift administrator, I want to be able to make changes to the
+  bridge interface using the same tools I do on the other interfaces.
+
+* As an OpenShift developer, I want a mechanism for bridge configuration that
+  is well-tested and less likely to break than a Bash script.
+
+* As an OpenShift developer, I want to simplify the configure-ovs.sh script
+  by providing a better way to do advanced bridge configuration.
+
+### Goals
+
+Once this design is implemented, deployers of OpenShift will be able to
+explicitly configure br-ex to their exact specifications and will be able to
+modify it after deployment using standard networking tools.
+
+### Non-Goals
+
+Complete replacement of configure-ovs. This mechanism is intended for more
```

Comment:
What are the cases where configure-ovs.sh would still be used?

Reply:
Probably most single-nic deployments where the logic for creating br-ex is simple. I don't anticipate this new mechanism being widely used in AWS, Azure, or GCP, for instance. In fact, I'm not sure how it will work in non-on-prem envs.

Reply:
Agreed. Can we add that as an example to this doc?

Reply:
Sure, done.